plugins/strategy/action.go

```diff
@@ -62,13 +58,59 @@ func (a *Action) CapCount(min, max int64) {
 
 	if newCount != oldCount {
 		a.Meta[metaKeyCountCapped] = true
+		r := fmt.Sprintf("capped count from %d to %d to stay within limits", oldCount, newCount)
+		a.handleCountChange(oldCount, newCount, r)
+	}
+}
+
+// LimitChange is used to modify the desired action based on the operator
+// specified maximum allowed change.
+func (a *Action) LimitChange(currentCount, maxChange int64) {
+
+	oldCount, newCount := a.Count, a.Count
+
+	// Determine whether this action is scaling in or out so we can correctly
+	// perform our calculations.
+	if a.Count < currentCount {
+		if change := currentCount - a.Count; change > maxChange {
+			newCount = currentCount - maxChange
+		}
+	} else if a.Count > currentCount {
+		if change := a.Count - currentCount; change > maxChange {
+			newCount = currentCount + maxChange
+		}
+	}
```

Comment:
```suggestion
	if a.Count < currentCount - maxChange {
		newCount = currentCount - maxChange
	} else if a.Count > currentCount + maxChange {
		newCount = currentCount + maxChange
	}
```
this is a little simpler, although the original approach (which computes and compares changes) has less risk of overflow as `abs(count)` approaches `2^31`

Reply:
note, numerical aspects aside, this is equivalent to the previous: `a.Count < currentCount - maxChange` can only happen if `a.Count < currentCount`. and the same for the other case, mutatis mutandis.